File: python/algo/sorting/selection_sort.py
import sys


def sort(items: list) -> None:
    for i in range(0, len(items)):
        min_index = i
        for j in range(i, len(items)):
            if items[j] < items[min_index]:
                min_index = j
        items[i], items[min_index] = items[min_index], items[i]


def main() -> None:
    items = [int(x) for x in sys.stdin.readline().split()]
    sort(items)
    print(" ".join(map(str, items)))


if __name__ == "__main__":
    main()

File: javascript/algo/sorting/selection_sort.js
const readline = require('readline')

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
})

rl.on('line', function (line) {
  const array = line.split(' ').map(Number)
  selectionSort(array)
  console.log(array.join(' '))
  process.exit(0)
})

function swap (array, i, j) {
  const tmp = array[i]
  array[i] = array[j]
  array[j] = tmp
}

function selectionSort (array) {
  for (let i = 0; i < array.length; i++) {
    let minIdx = i
    for (let j = i + 1; j < array.length; j++) {
      if (array[minIdx] > array[j]) {
        minIdx = j
      }
    }
    if (minIdx !== i) {
      swap(array, i, minIdx)
    }
  }
}

module.exports = selectionSort
